containers/src/Kokkos_UnorderedMap.hpp

```diff
@@ -413,23 +414,33 @@ class UnorderedMap {
   /// This is <i>not</i> a device function; it may <i>not</i> be
   /// called in a parallel kernel.
   bool rehash(size_type requested_capacity = 0) {
+    return rehash(execution_space{}, requested_capacity);
```

Comment:
We should probably fence after calling the overload taking an execution space instance.

Reply:
Do you agree that it's a `Kokkos::fence` that's needed?
```c++
bool rehash(size_type requested_capacity = 0) {
  bool result = rehash(execution_space{}, requested_capacity);
  Kokkos::fence("Kokkos::UnorderedMap::rehash: fence after rehasing");
  return result;
}
```

Reply:
That's what we are normally doing.